java/client/src/main/java/glide/api/models/commands/geospatial/GeoSearchOrigin.java

```diff
@@ -0,0 +1,61 @@
+/** Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0 */
+package glide.api.models.commands.geospatial;
+
+import glide.api.commands.GeospatialIndicesBaseCommands;
+import lombok.NonNull;
+import org.apache.commons.lang3.ArrayUtils;
+
+/**
+ * The query's starting point for {@link GeospatialIndicesBaseCommands} command.
+ *
+ * @see <a href="https://redis.io/commands/geosearch/">redis.io</a>
+ */
+public final class GeoSearchOrigin {
```

Comment:
As an option, you can have `GeoSearchOrigin` as an interface, and make two classes which implement that. See `KeysOrWeightedKeys` for example.